src/Symfony/Component/Routing/UrlHelper.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * A helper service for manipulating URLs within and outside the request scope.
+ *
+ * @author Valentin Udaltsov <udaltsov.valentin@gmail.com>
+ */
+final class UrlHelper
```

Comment:
Alternative names: `UrlManipulator`, `UrlBuilder`

Reply:
`UrlManipulator` is nice. Let's see what other reviewers think.
`UrlBuilder` is bad as a stateless service name.